olm/src/jsMain/kotlin/io/github/matrixkt/olm/PkDecryption.kt

```diff
@@ -0,0 +1,92 @@
+package io.github.matrixkt.olm
+
+import org.khronos.webgl.Uint8Array
+import kotlin.random.Random
+
+public actual class PkDecryption {
+    private val ptr: JsOlm.PkDecryption
+    public actual val publicKey: String
+
+    private constructor(ptr: JsOlm.PkDecryption, publicKey: String) {
+        this.ptr = ptr
+        this.publicKey = publicKey
+    }
+
+    private constructor(ptr: JsOlm.PkDecryption) {
+        this.ptr = ptr
+        this.publicKey = ""
+    }
+
+    public actual constructor(random: Random) {
+        ptr = JsOlm.PkDecryption()
+        try {
+            this.publicKey = ptr.generate_key()
+        } catch (e: Exception) {
+            clear()
+            throw e
+        }
+    }
+
+    public actual fun clear() {
+        ptr.free()
+    }
+
+    public actual val privateKey: ByteArray
+        get() {
+            val key: Uint8Array = ptr.get_private_key()
+            val t: String = key.toString()
+            return t.encodeToByteArray()
```

Comment:
What does `Uint8Array.toString` do?

Reply:
not what i thought it does. i need to find a way to cast it safely. actually the only thing i found is to iterate over the items and cast uints to ints and store it in a byteArray

Reply:
now casts correctly.